music/src/core/recorder.ts

```diff
@@ -72,27 +72,47 @@ export class Recorder {
     const inputs = midi.inputs.values();
     for (let input = inputs.next(); input && !input.done;
          input = inputs.next()) {
-      const midiInput = input.value;
-      midiInput.onmidimessage = (event) => {
-        this.midiMessageReceived(event);
-      };
+      this.midiInputs.push(input.value);
     }
   }
 
   isRecording() {
     return this.recording;
   }
 
+  /**
+   * Returns a list of all the currently connected MIDI inputs.
```

Comment:
Are these the ones we are recording? We also need a (static?) method to tell us what inputs are available so that users can choose the one they want. Or is this just easy enough for them to do themselves with WebMIDI?

Reply:
No, these are all of the ones that are connected (i.e. not recording necessarily). I think this is the list you're talking about. (If inputs aren't "connected" they're not seen by the MidiAccess interface)

Reply:
so you could do like `recorder.start(recorder.getMIDIInputs[0])`